### PR TITLE
Add content-format and lwm2m version to registration request 

### PR DESCRIPTION
**Pull request by rbtchc**

### CODE REVIEW

**Comment by rbtchc** [comment]
1. Add content-format and lwm2m version to registration request
1. Suggest to support OMA TLV format by default. It's a MUST according to LWM2M specification

**Comment by mike-scott** [comment]
Hi Robert,

I like these changes, however, I'm continuing to work on the "new" branch and I'll wait till I can pull these in on top of my work prior to submitting to Zephyr for RFC PR.  Is that ok?

File: samples/net/lwm2m_client/src/lwm2m-client.c [hunk 1]
```diff
@@ -273,7 +273,7 @@ void main(void)
 
 	k_sem_take(&quit_lock, K_FOREVER);
 
-#if defined(CONFIG_NET_IPV6)
+#if defined(CONFIG_NET_IPV4)
```
Comment (mike-scott) [inline samples/net/lwm2m_client/src/lwm2m-client.c]: This fix has been mentioned a few times.  I'd like to make the correction in my branch as it's part of review.

**Review by mike-scott — COMMENTED** [review]
I'd like to fold this back into my original patches.  It was also mentioned during review.  Is that ok?

**Comment by rbtchc** [comment]
Sure. Just integrate the ones you think that's useful in your patches.

**Pull request closed without merging**